src/ensembl/src/content/app/species/components/species-main-view/SpeciesMainViewStats.tsx

```diff
@@ -0,0 +1,217 @@
+
+import styles from './SpeciesMainView.scss';
+
+type Props = {
+  activeGenomeId: string | null;
+  genomeStats: GenomeStats | undefined;
+  exampleFocusObjects: ExampleFocusObject[];
+  fetchStatsForActiveGenome: () => void;
+};
+
+const ExampleLinks = (props: {
+  exampleLinks?: Partial<urlObj>;
+  exampleLinkText?: string;
+}) => {
+  const { exampleLinks, exampleLinkText } = props;
+
+  const [showPointerBox, setShowPointerBox] = useState(false);
+  const anchorRef = useRef<HTMLSpanElement>(null);
+  const containerRef = useRef<HTMLSpanElement>(null);
+
+  if (!exampleLinks) {
+    return null;
+  }
+
+  return (
+    <div className={styles.exampleLink}>
+      <span ref={containerRef}>
+        <span
+          className={styles.exampleLinkText}
+          ref={anchorRef}
+          onClick={() => setShowPointerBox(!showPointerBox)}
+        >
+          {exampleLinkText}
+          {showPointerBox && (
+            <PointerBox
+              anchor={anchorRef.current as HTMLDivElement}
+              onOutsideClick={() => setShowPointerBox(false)}
+              position={Position.BOTTOM_RIGHT}
+              container={containerRef.current}
+              autoAdjust={true}
+              classNames={{
+                box: styles.pointerBox,
+                pointer: styles.pointerBoxPointer
+              }}
+            >
+              <ViewInApp links={exampleLinks} />
+            </PointerBox>
+          )}
+        </span>
+      </span>
+    </div>
+  );
+};
+
+const getCollapsedContent = (statsSection: StatsSection) => {
+  const { primaryStats, secondaryStats, section, exampleLinks } = statsSection;
+  const { title, exampleLinkText } = sectionGroupsMap[section];
+
+  return (
+    <div className={styles.collapsedContent}>
+      <span className={styles.title}>{title}</span>
+      {primaryStats && (
+        <div className={styles.primary}>
+          <span className={styles.primaryValue}>
+            {primaryStats.primaryValue}
+          </span>
+          <span className={styles.primaryUnit}>{primaryStats.primaryUnit}</span>
+        </div>
+      )}
+
+      {secondaryStats && (
+        <div className={styles.secondary}>
+          <span className={styles.secondaryValue}>
+            {secondaryStats.secondaryValue}
+          </span>
+          <span className={styles.secondaryUnit}>
+            {secondaryStats.secondaryUnit}
+          </span>
+        </div>
+      )}
+
+      <ExampleLinks
+        exampleLinks={exampleLinks}
+        exampleLinkText={exampleLinkText}
+      />
+    </div>
+  );
+};
+
+const getExpandedContent = (statsSection: StatsSection) => {
+  const { groups, exampleLinks, section } = statsSection;
+  const { exampleLinkText } = sectionGroupsMap[section];
+  return (
+    <div className={styles.expandedContent}>
+      {groups.map((group, group_index: number) => {
+        const { title, stats } = group;
+        const primaryStatsClassName = classNames(styles.primaryStats, {
```

Comment:
Am I reading this correctly and the calculation of this class name does not depend on anything happening inside the loop? If so, then it should be calculated outside the `.map`

Reply:
That's done